### PR TITLE
Add CRD Ecal geometry

### PR DESCRIPTION
**Pull request by qwert2333**

### CODE REVIEW

**Comment by mirguest** [comment]
Hi @qwert2333 , 
Please add detail description of your geometry in this PR. 
If possible, you could upload a snapshot of the ECAL.

Tao

File: Detector/DetCRD/compact/ecalBarrel.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  
+  <readouts>
+    <readout name="CaloHitsCollection">
+      <!-- <segmentation type="NoSegmentation"/> -->
+
+      <!--segmentation type="CartesianGridXYZ"
+                    grid_size_x="1*cm"
+                    grid_size_y="1*cm"
+                    grid_size_z="1*cm"/-->
+      <id>block:8,layer:12,piece:12,bar:12</id>
```
Comment (mirguest) [inline Detector/DetCRD/compact/ecalBarrel.xml:41]: @qwert2333 @LinghuiWuIHEP

Please note that you should have `system` field defined in this ID definition.

Tao

**Comment by qwert2333** [comment]
Detector description(DD4hep): 
  First version v0r0: full BGO crystal in all detector. 
  Geometry: 
    Global (could be modified in ecalBarrel.xml)
        Inner radius: 1800mm. 
        Z length: 4600mm. 
        Thickness: 280mm. ( ! Must be integer multiple of 10).
    Crystal bars placement: 
        Bar size:  1cm*1cm*~40cm  ( 1cm*1cm is hard-coding in CRDEcal.cpp ).
        Odd layer: along with phi direction, 3~5 bars in each layer.     ( ! Number of bars is Hard-coding )
        Even layer: along with Z axis, 12 bars in layer.    ( ! Number of bars is fixed to 12) 
  Cell ID: 
    system: 4 bit.  (8 parts)  
    layer: 6 bit    (~28 layers)
    block: 8bit    (<200 bars in phi direction)
    bar: 10 bit     ( max~460 bars in Z direction)

File: Detector/DetCRD/src/Calorimeter/CRDEcal.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+using dd4hep::rec::LayeredCalorimeterData;
+using namespace dd4hep;
+using namespace std;
+static dd4hep::Ref_t create_detector(dd4hep::Detector& theDetector,
+                                     xml_h e,
+                                     dd4hep::SensitiveDetector sens) {
+
+    xml_det_t x_det = e;
+
+    std::string det_name = x_det.nameStr();
```
Comment (mirguest) [inline Detector/DetCRD/src/Calorimeter/CRDEcal.cpp:31]: In CRDEcal.cpp, please make sure Tabs and Spaces are not mixed.
Please consider using 4 space for indentation.

**Review by mirguest — CHANGES_REQUESTED** [review]
Please make sure your code is well format.

**Comment by fucd** [comment]
Hi @qwert2333 

My suggestion is that don't use Tube to create envelope, but PolyhedraRegular. Because to put other detector components into space between  PolyhedraRegular and Tube is possibly required. 
`dd4hep::Tube		envelope(R0, (R0+h0)/cos(22.5*degree), Z0/2, 0, 2*M_PI);`

Chengdong

**Pull request merged**